v1/transform.go

```diff
@@ -0,0 +1,228 @@
+			DeviceResource: ro.DeviceResource,
+			Mappings:       ro.Mappings,
+		}
+	}
+	return operations
+}
+
+// TransformToV2DeviceProfile transform v1 profile to v2
+func TransformToV2DeviceProfile(profile v1Models.DeviceProfile) models.DeviceProfile {
+	dto := models.DeviceProfile{
+		Description:     profile.Description,
+		Name:            profile.Name,
+		Manufacturer:    profile.Manufacturer,
+		Model:           profile.Model,
+		Labels:          profile.Labels,
+		DeviceResources: nil,
+		DeviceCommands:  nil,
+	}
+	dto.DeviceResources = toV2DeviceResources(profile.DeviceResources)
+	dto.DeviceCommands = toV2DeviceCommands(profile.DeviceCommands)
+	return dto
+}
+
+func toV2DeviceResources(deviceResources []v1Models.DeviceResource) []models.DeviceResource {
+	resources := make([]models.DeviceResource, len(deviceResources))
+	for i, r := range deviceResources {
+		resources[i] = models.DeviceResource{
+			Description: r.Description,
+			Name:        r.Name,
+			IsHidden:    false,
+			Tags:        toV2Tags(r.Tags),
+			Properties: models.ResourceProperties{
+				ValueType:    r.Properties.Value.Type,
+				ReadWrite:    r.Properties.Value.ReadWrite,
+				Units:        r.Properties.Units.DefaultValue,
+				Minimum:      r.Properties.Value.Minimum,
+				Maximum:      r.Properties.Value.Maximum,
+				DefaultValue: r.Properties.Value.DefaultValue,
+				Mask:         r.Properties.Value.Mask,
+				Shift:        r.Properties.Value.Shift,
+				Scale:        r.Properties.Value.Scale,
+				Offset:       r.Properties.Value.Offset,
+				Base:         r.Properties.Value.Base,
+				Assertion:    r.Properties.Value.Assertion,
+				MediaType:    r.Properties.Value.MediaType,
+			},
+			Attributes: toV2Attributes(r.Attributes),
+		}
+	}
+
+	return resources
+}
+
+func toV2Tags(tags map[string]string) map[string]interface{} {
+	dto := make(map[string]interface{})
+	for k, v := range tags {
+		dto[k] = v
+	}
+	return dto
+}
+
+func toV2Attributes(attributes map[string]string) map[string]interface{} {
+	dto := make(map[string]interface{})
+	for k, v := range attributes {
+		dto[k] = v
+	}
+	return dto
+}
+
+func toV2DeviceCommands(deviceCommands []v1Models.ProfileResource) []models.DeviceCommand {
+	commands := make([]models.DeviceCommand, len(deviceCommands))
+	for i, c := range deviceCommands {
+		commands[i] = models.DeviceCommand{
+			Name:               c.Name,
+			IsHidden:           false,
+			ReadWrite:          "",
+			ResourceOperations: nil,
+		}
+		if len(c.Get) > 0 && len(c.Set) > 0 {
+			commands[i].ReadWrite = common.ReadWrite_RW
+		} else if len(c.Set) > 0 {
+			commands[i].ReadWrite = common.ReadWrite_W
+		} else {
+			commands[i].ReadWrite = common.ReadWrite_R
+		}
+		var ros []models.ResourceOperation
+		for _, op := range c.Get {
+			ro := models.ResourceOperation{
+				DeviceResource: op.DeviceResource,
+				DefaultValue:   "",
+				Mappings:       op.Mappings,
+			}
+			ros = append(ros, ro)
+		}
+		for _, op := range c.Set {
```

Comment:
if you just combine Get and Set, the resourceOperation will be duplicated

Reply:
Modified, 
- if the permission is RW,  matching the get and set operation for creating v2 operation.
- if the permission is W, using set operation
- if the permission is R, using get operation